firedrake/interpolation.py

```diff
@@ -747,7 +567,7 @@ def _interpolate(
             # don't have to worry about skipping over missing points here
             # because I'm going from the input ordering VOM to the original VOM
             # and all points from the input ordering VOM are in the original.
-            interp = action(expr_adjoint(self.to_input_ordering_interpolate), f_src_at_dest_node_coords_dest_mesh_decomp)
+            interp = action(expr_adjoint(self.point_eval_input_ordering), f_src_at_dest_node_coords_dest_mesh_decomp)
```

Comment:
Do we need an adjoint here? Could swapping the arguments to action not work?
```suggestion
            interp = action(f_src_at_dest_node_coords_dest_mesh_decomp, self.point_eval_input_ordering)
```